pkg/nftables/networkpolicy.go

```diff
@@ -60,6 +60,19 @@ func clusterwideNetworkPolicyIngressRules(np firewallv1.ClusterwideNetworkPolicy
 	return uniqueSorted(rules)
 }
 
+func clusterwideNetworkPolicyEgressDNSCacheRules(cache FQDNCache, logAcceptedConnections bool) (nftablesRules, error) {
+	addr, err := cache.CacheAddr()
+	if err != nil {
+		return nil, err
+	}
+	base := []string{"ip saddr == @cluster_prefixes", fmt.Sprintf("ip daddr { %s }", addr)}
+	comment := fmt.Sprintf("accept traffic for dns cache")
```

Comment:
To clarify a bit more, I think this rule is solely to be able to retrieve the intercepted traffic DNS traffic by the DNAT rules, right? So maybe this helps to point that out:

```suggestion
	comment := fmt.Sprintf("accept intercepted traffic for dns cache")
```